byway/src/main/webapp/script.js

```diff
@@ -12,19 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- *
- */
-/* exported addRandomGreeting */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function initMap() {
+  var directionsService = new google.maps.DirectionsService();
+  var directionsRenderer = new google.maps.DirectionsRenderer();
+  var start = new google.maps.LatLng(37.7699298, -122.4469157);
+  var end = new google.maps.LatLng(37.7683909618184, -122.51089453697205);
+  var mapOptions = {
+    zoom: 14,
+    center: start
+  }
+  var map = new google.maps.Map(document.getElementById('map'), mapOptions);
+  directionsRenderer.setMap(map);
+     document.getElementById("route").addEventListener("click", function() {
+    calcRoute(directionsService, directionsRenderer, start, end);
+  });
+}
+
+function calcRoute(directionsService, directionsRenderer, start, end) {
+  console.log('getting route');
+  var request = {
+      origin:  start,
+      destination: end,
+      travelMode: 'DRIVING'
+  };
+  directionsService.route(request, function(response, status) {
+    if (status == 'OK') {
+      directionsRenderer.setDirections(response);
+    } else {
+      window.alert("Could not calculate route due to: " + status);
+    }
+  });
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function foo(){
```

Comment:
Probably want to get rid of this before committing to master

Reply:
Okay, I was just using it to mock button onclick, but I'll remove.

Reply:
I see. It may be more useful to stub out the real function and log a message there instead. This way it is more clear that it is a place holder for debugging.

Reply:
done!